src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -413,7 +413,6 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 		this.blockModelContentChange = true;
 		try {
 			this.updateTextEditorModel(value);
-			this.setDirty(false);
```

Comment:
@alexandrudima I'm pretty sure this change was necessary. We need the model to know that this is now the current saved state.

If after reloading the file on disk, if you press undo it shows that the file is still saved, after a redo it shows the file is unsaved, and if you attempt to close the file it prompts you to save it, even though it is up to date.

Here's a gif where I modify the file with vim, press undo, then redo: 
![dirty](https://user-images.githubusercontent.com/1206952/29886776-80176122-8d78-11e7-96f6-eb469b128a2b.gif)

Reply:
@stringham thanks we are looking into restoring that, you are right, it is needed.

Reply:
@stringham :+1: thank you :heart: !